core/renderers/geras/info.js

```diff
@@ -339,6 +340,41 @@ Blockly.geras.RenderInfo.prototype.getSpacerRowHeight_ = function(prev, next) {
   return this.constants_.MEDIUM_PADDING;
 };
 
+/**
+ * @override
+ */
+Blockly.geras.RenderInfo.prototype.alignRowElements_ = function() {
+  for (var i = 0, row; (row = this.rows[i]); i++) {
+    if (row.hasStatement) {
+      var statementInput = row.getLastInput();
+      var currentWidth = row.width - statementInput.width;
+      var desiredWidth = this.statementEdge - this.startX;
+    } else {
+      var currentWidth = row.width;
+      var desiredWidth = this.width - this.startX;
+      if (Blockly.blockRendering.Types.isBottomRow(row) && row.hasFixedWidth) {
+        desiredWidth = this.constants_.MAX_BOTTOM_WIDTH;
+      }
+    }
+    var missingSpace = desiredWidth - currentWidth;
+    if (missingSpace) {
+      this.addAlignmentPadding_(row, missingSpace);
+    }
+  }
+};
+
+/**
+ * @override
+ */
+Blockly.geras.RenderInfo.prototype.getSpacerRowWidth_ = function(prev, next) {
+  // The width of the spacer before the bottom row should be the same as the
+  // bottom row.
+  if (Blockly.blockRendering.Types.isBottomRow(next) && next.hasFixedWidth) {
```

Comment:
I feel like the reason we need to do this is because the spacer is added after computeBounds. If the bottom row width is fixed, why does it matter what the spacer width is.